### PR TITLE
Added: Enable Download Client Priorities

### LINKED ISSUE #229: Re-implement Priority in download clients
(Maybe we don't need to, does anyone find this useful for adding movies to your download clients (the ones that support it)?)

Definitely planning on implementing this, once I get around to it.

### PR DESCRIPTION
**Pull request by Qstick**
#### Database Migration
NO

#### Description

Re-enable Download Client Priorities

#### Issues Fixed or Closed by this PR

Fixes #229

### CODE REVIEW

File: src/NzbDrone.Core/Parser/Model/RemoteMovie.cs [hunk 1]
```diff
@@ -12,6 +12,16 @@ public class RemoteMovie
         public Movie Movie { get; set; }
         public MappingResultType MappingResult { get; set; }
 
+        public bool IsRecentMovie()
```
Comment (galli-leo) [inline src/NzbDrone.Core/Parser/Model/RemoteMovie.cs]: IMO this makes much more sense being on the actual movie object.

File: src/UI/Form/CheckboxTemplate.hbs [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <div class="col-sm-5">
         <div class="input-group">
             <label class="checkbox toggle well">
-                <input type="checkbox" name="fields.{{order}}.value"/>
+                <input type="checkbox" name="fields.{{order}}.value" validation-name="{{name}}"/>
```
Comment (galli-leo) [inline src/UI/Form/CheckboxTemplate.hbs]: Why did you add this?

File: src/NzbDrone.Core/Parser/Model/RemoteMovie.cs (hunk 1 above)
Comment (Qstick) [reply src/NzbDrone.Core/Parser/Model/RemoteMovie.cs]: I kinda agree, I just added where Sonarr had it.

Comment (galli-leo) [reply src/NzbDrone.Core/Parser/Model/RemoteMovie.cs]: Ah interesting, I would still move it to the movie itself (and also make it a property). Maybe we need it somewhere else in the future.

**Comment by Qstick** [comment]
@galli-leo, I moved the method over to Movie.cs, however I don't know that it makes since for this to be a property instead of a method, we would have to constantly recalculate it as it would change daily. Seems to make more sense to leave as a method instead. 

Additionally, I noticed many of my older movies only had a cinema date and not a physical release so I added a backup to the logic. If physical release is in the last 21 days then isRecent, if cinema date is in the last 120 days then isRecent. If neither Physical or Cinema have a date (I don't know that this is even possible, but good to have backup case) then isRecent. Tested on a few nzbs and priority is set as intended.

**Comment by danjames92** [comment]
Wouldn't it make more sense to fix the bug with the priorities first and then make any changes to the priorities later in a separate PR?

I'm not sure why the behaviour needs to the change as such.

**Comment by galli-leo** [comment]
@Qstick I think the 120 days from cinemas is a good option.

I still think it should be a property. Don’t really understand how we would have to recalculate it daily? Also isAvailable works the same and since it’s just data, a property makes more sense (imo).

I know, I am being pedantic, but release is gonna have to wait anyways 😅

**Comment by Qstick** [comment]
@galli-leo is available is setup same as I have this setup. 

https://github.com/Radarr/Radarr/blob/25e10e26e308b4ad41dcaa428e4a171413b2ae08/src/NzbDrone.Core/Movies/Movie.cs#L77

It's a method for the movie model, but a property for the API resource that calls back to the model method. Is that what you mean?

**Comment by galli-leo** [comment]
@Qstick Ah yeah forgot about the delay. But yeah I would make isRecent a property like in the API model, since we don’t need a parameter like with isAvailable.
